### PR TITLE
Sends correctly formatted params to the backend for notes

### PR DESCRIPTION
**Pull request by sharonwarner**
Sending null to the backend causes an Action::Controller empty parameter error. We were accidentally sending that to VACOLS notes creation. Sending false fixes this problem.

Connects #2292 

Set a breakpoint in self.update_vacols_after_dispatch! in Fakes::AppealRepository

- [ ] Confirm that vacols_note is false when no special issues are selected
- [ ] Confirm that vacols_note is true when no special issues that require notes are selected (National Cemetery Administration)

### CODE REVIEW

File: client/app/containers/EstablishClaimPage/EstablishClaim.jsx [hunk 1]
```diff
@@ -330,9 +330,9 @@ export default class EstablishClaim extends BaseForm {
     });
 
     // We want to trim the vacols note to 280 char. As that is
-    // a DB column constriant
+    // a DB column constraint
```
Comment (NickHeiner) [inline client/app/containers/EstablishClaimPage/EstablishClaim.jsx]: 👍

File: client/app/containers/EstablishClaimPage/EstablishClaim.jsx [hunk 2]
```diff
@@ -330,9 +330,9 @@ export default class EstablishClaim extends BaseForm {
...
     let data = ApiUtil.convertToSnakeCase({
-      vacolsNote: (vacolsNote && vacolsNote.substring(0, 280))
+      vacolsNote: (vacolsNote) ? vacolsNote.substring(0, 280) : false
```
Comment (NickHeiner) [inline client/app/containers/EstablishClaimPage/EstablishClaim.jsx]: Why do we need parens around `vacolsNote`?

Comment (NickHeiner) [inline client/app/containers/EstablishClaimPage/EstablishClaim.jsx]: Why was this change necessary?

File: client/test/karma/util/ApiUtil-test.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { expect } from 'chai';
+import ApiUtil from '../../../app/util/ApiUtil';
+
+describe('ApiUtil', () => {
+  context('.convertToSnakeCase', () => {
+
+    let camelCaseObject = {
```
Comment (NickHeiner) [inline client/test/karma/util/ApiUtil-test.js]: Please default to `const` instead of `let`. `let` should only be used for mutable variables.

File: client/test/karma/util/ApiUtil-test.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    let snakeCaseObject = {
+      vacols_id: null,
+      vacols_name: {
+        first_name: 'Jane',
+        last_name: 'Smith'
+      }
+    };
+
+    it('returns a correctly formatted object', () => {
+      expect(ApiUtil.convertToSnakeCase(camelCaseObject)).to.eql(snakeCaseObject);
```
Comment (NickHeiner) [inline client/test/karma/util/ApiUtil-test.js]: I prefer `.deep.equal` because it's more explicitly readable than `eql`.

File: client/app/util/ApiUtil.js [hunk 5]
```diff
@@ -7,8 +7,8 @@ import StringUtil from './StringUtil';
 const ApiUtil = {
 
   // Converts camelCase to snake_case
-  convertToSnakeCase(data = {}) {
-    if (typeof data !== 'object') {
+  convertToSnakeCase(data) {
+    if (typeof data !== 'object' || data === null) {
```
Comment (NickHeiner) [inline client/app/util/ApiUtil.js]:
```
> typeof null
'object'
```

God I love JS.

Comment (NickHeiner) [inline client/app/util/ApiUtil.js]: We could also use `_.isObject` or `_.isObjectLike` here. But it's up to you.

https://lodash.com/docs/4.17.4#isObject

File: client/test/karma/util/ApiUtil-test.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
+import { expect } from 'chai';
+import ApiUtil from '../../../app/util/ApiUtil';
+
+describe('ApiUtil', () => {
+  context('.convertToSnakeCase', () => {
+
+    let camelCaseObject = {
+      vacolsId: null,
```
Comment (NickHeiner) [inline client/test/karma/util/ApiUtil-test.js]: Just for fun, want to add a few test cases for other sorts of fields?

```js
    let camelCaseObject = {
      vacolsId: null,
      falseField: false,
      undefinedField: undefined,
      emptyStringField: '',
      vacolsName: {
        firstName: 'Jane',
        lastName: 'Smith'
      }
    };
```

File: client/app/util/ApiUtil.js (hunk 5 above)
Comment (sharonwarner) [inline client/app/util/ApiUtil.js]: Lol.
